src/lsp/sql_parser/grammar.mly

```diff
@@ -0,0 +1,145 @@
+(*                                                                        *)
+(*                        SuperBOL OSS Studio                             *)
+(*                                                                        *)
+(*  Copyright (c) 2022-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(* All rights reserved.                                                   *)
+(* This source code is licensed under the GNU Affero General Public       *)
+(* License version 3 found in the LICENSE.md file in the root directory   *)
+(* of this source tree.                                                   *)
+(*                                                                        *)
+(**************************************************************************)
+open Sql_ast
+open Cobol_common.Srcloc.INFIX
+module Overlay_manager =
+  Cobol_preproc.Src_overlay.New_manager (struct
+    let name = __MODULE__
+  end) ()
+
+%}
+%token EOF
+%token EXEC SQL END_EXEC
+%token BEGIN END DECLARE SECTION DISCONNECT COMMA LPAR RPAR DOT
+(*For whenever*)
+%token WHENEVER NOT FOUND SQLERROR SQLWARNING CONTINUE PERFORM GOTO
+(*For include*)
+%token INCLUDE SQLCA SQLDA
+(*For Connect*)
+%token CONNECT TO USER WITH NO PROMPT RETURNING IDENTIFIED BY AT USING RESET DSN AS
+(*For Transaction*)
+%token START TRANSACTION
+(*For Rollback*)
+%token ROLLBACK WORK RELEASE SAVEPOINT 
+(*For commit*)
+%token COMMIT
+(*For SQL*)
+%token ASTERIX SELECT FROM
+%token <string> WORD
+%token <string> STRING
+%token <string> DIGITS
+%token <string> COBOL_VAR
+%token <string> BACKSLASHED_VAR
+%start <esql_instuction> main 
+
+%%
+
+
+let loc (X) ==
+  | x = X; { x &@ Overlay_manager.join_limits $sloc }
+
+  
+let main :=
+| EXEC; SQL; stm=esql ; END_EXEC; EOF; {stm}
+
+let variable := 
+| s=WORD; {SqlVar s} 
+| s=COBOL_VAR; {CobolVar s} 
+
+let esql := 
+| AT; v=loc(variable); stm=esql; {At(v, stm)}
+| BEGIN; DECLARE; SECTION; {BeginDeclare}
+| END; DECLARE; SECTION; {EndDeclare}
+| START; TRANSACTION; {StartTransaction}
+| WHENEVER; c=whenever_condtion; k= whenever_continuation; {Whenever(c, k)}
+| i = sql; {Sql(i)}
+| INCLUDE; i=include_stm; {Include(i)}
+| CONNECT; c=connect_stm; {Connect(c)}
+| ROLLBACK; r=rb_work_or_tran; a=rb_args; {Rollback(r, a)}
+| COMMIT; wt= option(rb_work_or_tran); RELEASE; {Commit(wt, true)}
+| COMMIT; wt= option(rb_work_or_tran); {Commit(wt, false)}
+| SAVEPOINT; s=loc(variable); {Savepoint(s)}
+
+let rb_work_or_tran :=
+| WORK; {Work}
+| TRANSACTION; {Transaction}
+
+let rb_args :=
+| RELEASE; {Release}
+| TO; SAVEPOINT; v = loc(variable); {To(v)}
+| TO; v = loc(variable); {To(v)} 
+
+let connect_stm :=
+/*
+EXEC SQL CONNECT TO :dbname [ AS :db_conn_id ] 
+USER :username USING :db_data_source 
+IDENTIFIED BY :password 
+*/
+| TO; dbname= loc(variable); db_conn_id=option(as_var); USER; username=loc(variable);
```

Comment:
Actually, database name, username and password may be specified as string literals.